C301.tex

```diff
@@ -68,24 +71,143 @@
 % and used to generate entries in the ASCL Index at the end of the Proceedings volume.
 % The ascl.py command will scan your paper on possible code names.
 % Don't leave these in! - replace them with ones relevant to your paper.
-%\ooindex{FOOBAR, ascl:1101.010}
 
+\section{Docs like code}
+
+% - Enable the team to write Docs
+% - Docs like code fits our team's workflow.
+% - Reference Docs
+% - Exploration of new technologies
+
+At Rubin Observatory \citep{2019ApJ...873..111I}, every team member participates in documentation.
+To enable anyone to be a successful documentarian, we invested in creating a documentation platform that is simple and convenient to author and publish in.
+Although there are many valid frameworks for creating and publishing documentation, at Rubin we were attracted to the \emph{docs-like-code} \citep{Gentle2022} framework that handles documentation content as plain text files in version-controlled source repositories.
+As a software development organization, particularly within the Rubin Data Management subsystem, a docs-like-code approach offers the best developer experience because it fits into our existing development workflows.
+Code editors, GitHub pull requests, and Git branches and tags work equally well for code development as they do for docs development.
+
+We have found benefits of treating docs-like-code beyond the streamlined workflow.
+Since code repositories or other API-accessible systems contain the source-of-truth for many of the things we document, a docs-like-code framework facilitates building reference documentation directly from that source-of-truth.
+This is commonly done for building Python API documentation.
+At Rubin, we have also built custom reference generators (for example, documentation for our Kubernetes deployments that are deployed from Git, \url{https://phalanx.lsst.io}).
+% , pipelines, and services interfaces (using FastAPI where the data models for endpoints not only validate requests and responses, but also yield OpenAPI documentation).
+A docs-like-code system also encourages the development of tooling (which is discussed in this work), and we have taken the opportunity to use the documentation platform as a low-risk testbed for new technologies like Kubernetes and GitOps, Kafka event streaming, and Python web service development patterns.
+This work describes the key components of the Rubin documentation platform, and their relationship to the overall user experience.
+
+\section{Documentation deployment}
+
+% Automatic deployment
+% Features beyond read the docs
+
+At Rubin, we created our own documentation deployment system, \textit{LSST the Docs} \citep{SQR-006}, to deploy and host our documentation.
+This system is designed so that documentation deployments happen automatically in the course of developing a Git repository, without any extra effort from the developer.
+Each documentation project is hosted at its own subdomain of \href{https://www.lsst.io}{lsst.io}.
```

Comment:
Nit: It looked like www.lsst.io is now a redirect to lsst.io, so I've been changing URLs accordingly.

Reply:
Oh, that was a configuration issue when doing to temporary deployment while I can get the new Phalanx deployment figured out. The canonical URL is www.lsst.io again, and the apex lsst.io redirects back to that. I hope you haven't been updating a lot of codebases with that?

Reply:
Oh, whoops, thanks. No, I think it was just Phalanx and maybe one other repository. I'll switch Phalanx back and try to remember where else I did that.